governance/program/src/state.rs

```diff
@@ -0,0 +1,237 @@
+    pub no_vote_mint: Pubkey,
+
+    /// Used to validate signatory tokens in a round trip transfer
+    pub signatory_validation: Pubkey,
+
+    /// Used to validate admin tokens in a round trip transfer
+    pub admin_validation: Pubkey,
+
+    /// Used to validate voting tokens in a round trip transfer
+    pub vote_validation: Pubkey,
+
+    /// Source Token Holding account
+    pub source_holding: Pubkey,
+
+    /// Source Mint - either governance or council mint from Governance
+    pub source_mint: Pubkey,
+}
+
+/// Proposal state
+#[derive(Clone)]
+pub struct ProposalState {
+    /// Governance account type
+    pub account_type: GovernanceAccountType,
+
+    /// Proposal account
+    pub proposal: Pubkey,
+
+    /// Current status of the proposal
+    pub status: ProposalStateStatus,
+
+    /// Total signatory tokens minted, for use comparing to supply remaining during draft period
+    pub total_signing_tokens_minted: u64,
+
+    /// Link to proposal's description
+    pub description_link: [u8; MAX_PROPOSAL_DESCRIPTION_LINK_LENGTH],
+
+    /// Proposal name
+    pub name: [u8; MAX_PROPOSAL_NAME_LENGTH],
+
+    /// When the Proposal ended voting - this will also be when the set was defeated or began executing naturally.
+    pub voting_ended_at: u64,
+
+    /// When the Proposal began voting
+    pub voting_began_at: u64,
+
+    /// when the Proposal entered draft state
+    pub created_at: u64,
+
+    /// when the Proposal entered completed state, also when execution ended naturally.
+    pub completed_at: u64,
+
+    /// when the Proposal entered deleted state
+    pub deleted_at: u64,
+
+    /// The number of the transactions already executed
+    pub number_of_executed_transactions: u8,
+
+    /// The number of transactions included in the proposal
+    pub number_of_transactions: u8,
+
+    /// Array of pubkeys pointing at Proposal Transactions, up to 5
+    pub transactions: [Pubkey; MAX_TRANSACTIONS],
+}
+
+/// What state a Proposal is in
+#[derive(Clone, Debug, PartialEq)]
+pub enum ProposalStateStatus {
+    /// Draft
+    Draft,
+
+    /// Taking votes
+    Voting,
+
+    /// Votes complete, in execution phase
+    Executing,
+
+    /// Completed, can be rebooted
+    Completed,
+
+    /// Deleted
+    Deleted,
+
+    /// Defeated
+    Defeated,
+}
+
+impl Default for ProposalStateStatus {
+    fn default() -> Self {
+        ProposalStateStatus::Draft
+    }
+}
+
+/// Governance Vote Record
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct GovernanceVoteRecord {
+    /// Governance account type
+    pub account_type: GovernanceAccountType,
+
+    /// Proposal account
+    pub proposal: Pubkey,
+
+    /// The user who casted this vote
+    pub voter: Pubkey,
+
+    /// How many votes were unspent
+    pub undecided_count: u64,
+
+    /// How many votes were spent yes
+    pub yes_count: u64,
+
+    /// How many votes were spent no
+    pub no_count: u64,
```

Comment:
If we remove these 3 voting mints this can be just enum Option
`pub enum VoteResult {
 Yes(u8),
 No(u8)
}`
I think there is no need to support simultaneous Yes and No votes and no vote will be None